tests/integration_tests/api/system_backend/test_health.py

```diff
@@ -2,30 +2,78 @@
 from unittest import TestCase
 
 from parameterized import parameterized, param
-
+from tests.utils import create_client
 from tests.utils.hvac_integration_test_case import HvacIntegrationTestCase
 
 
 class TestHealth(HvacIntegrationTestCase, TestCase):
+    enable_vault_ha = True
+
+    def tearDown(self):
+        # If one of our test cases left the Vault cluster sealed, unseal it here.
+        self.manager.unseal()
+        super(TestHealth, self).tearDown()
 
     @parameterized.expand([
         param(
             'default params',
         ),
+        param(
+            'unsealed standby node',
+            vault_addr='https://127.0.0.1:8199',
+            method='HEAD',
+            expected_status_code=429,
+            ha_required=True,
+        ),
+        param(
+            'sealed standby node',
+            vault_addr='https://127.0.0.1:8199',
+            method='GET',
+            expected_status_code=503,
+            seal_first=True,
+            ha_required=True,
+        ),
```

Comment:
For posterity, here is the output from these regression test without the fix being added here:

```
======================================================================
ERROR: Test the Health system backend class's "read_health_status" method [with label='unsealed standby node', method='HEAD', vault_addr='https://127.0.0.1:8199', expected_status_code=429, seal_first=False, ha_required=True].
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jeffrey.hogan/.virtualenvs/hvac/lib/python3.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/Users/jeffrey.hogan/workspace/hvac/tests/integration_tests/api/system_backend/test_health.py", line 70, in test_read_health_status
    method=method,
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/api/system_backend/health.py", line 59, in read_health_status
    url=api_path,
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/adapters.py", line 158, in head
    return self.request('head', url, **kwargs)
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/adapters.py", line 266, in request
    errors = response.json().get('errors')
  File "/Users/jeffrey.hogan/.virtualenvs/hvac/lib/python3.7/site-packages/requests/models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

======================================================================
ERROR: Test the Health system backend class's "read_health_status" method [with label='sealed standby node', method='GET', vault_addr='https://127.0.0.1:8199', expected_status_code=503, seal_first=True, ha_required=True].
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jeffrey.hogan/.virtualenvs/hvac/lib/python3.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/Users/jeffrey.hogan/workspace/hvac/tests/integration_tests/api/system_backend/test_health.py", line 70, in test_read_health_status
    method=method,
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/api/system_backend/health.py", line 66, in read_health_status
    json=params,
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/adapters.py", line 93, in get
    return self.request('get', url, **kwargs)
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/adapters.py", line 269, in request
    utils.raise_for_error(response.status_code, text, errors=errors)
  File "/Users/jeffrey.hogan/workspace/hvac/hvac/utils.py", line 43, in raise_for_error
    raise exceptions.VaultDown(message, errors=errors)
hvac.exceptions.VaultDown: {"initialized":true,"sealed":true,"standby":true,"performance_standby":false,"replication_performance_mode":"unknown","replication_dr_mode":"unknown","server_time_utc":1544189688,"version":"1.0.0"}
```

@qeqar: Are there any additional test cases we should add here to capture the issues you have outlined in #339?

Reply:
After re-reading the issue, I realized we should go ahead and check both GET and HEAD methods for both types of scenarios.... 16ef641

Reply:
Sorry i missed the question :-)

ok we test one in 4xx one in 5xx, i am not sure if we should do positive test 200 too?

the unsealed standby with head  should cover my finding in #339